pfcpiface/aether.go

```diff
@@ -0,0 +1,849 @@
+		log.Errorln(err)
+		return err
+	}
+
+	log.Tracef("proccesed IPLookup %v request %+v", method, msg)
+
+	return nil
+}
+
+func (a *aether) createDstMacUpdateModule(ctx context.Context, ip *net.IPNet, nhop net.HardwareAddr) (string, error) {
+	tmp := make([]byte, 8) // 64 bit
+	copy(tmp[2:], nhop)
+
+	arg := &pb.UpdateArg{Fields: []*pb.UpdateArg_Field{
+		{Offset: 0, Size: 6, Value: binary.BigEndian.Uint64(tmp)},
+	}}
+
+	moduleName := datapathIfaceName + "DstIP" + ip.String() + "DstEther" + nhop.String()
+
+	return moduleName, a.createModule(ctx, moduleName, "Update", arg)
+}
+
+func (a *aether) deleteDstMacUpdateModule(ctx context.Context, ip *net.IPNet, nhop net.HardwareAddr) (err error) {
+	moduleName := datapathIfaceName + "DstIP" + ip.String() + "DstEther" + nhop.String()
+
+	return a.deleteModule(ctx, moduleName)
+}
+
+func (a *aether) createModule(ctx context.Context, moduleName, moduleClass string, args proto.Message) error {
+	any, err := anypb.New(args)
+	if err != nil {
+		log.Error("Error marshalling the rule", args, err)
+		return err
+	}
+
+	req := &pb.CreateModuleRequest{
+		Name:   moduleName,
+		Mclass: moduleClass,
+		Arg:    any,
+	}
+
+	resp, err := a.client.CreateModule(ctx, req)
+
+	if err != nil {
+		log.Errorf("CreateModule RPC failed with err: %v\n", err)
+		return err
+	}
+
+	if resp.GetError() != nil && resp.GetError().Code != 0 {
+		log.Errorf("CreateModule request '%+v' failed with err: %v\n", req, resp.GetError())
+		return status.Error(codes.Code(resp.GetError().Code), resp.GetError().Errmsg)
+	}
+
+	log.Tracef("Created new module %v of type %v", moduleName, moduleClass)
+
+	return nil
+}
+
+func (a *aether) deleteModule(ctx context.Context, moduleName string) error {
+	req := &pb.DestroyModuleRequest{
+		Name: moduleName,
+	}
+
+	resp, err := a.client.DestroyModule(ctx, req)
+
+	if err != nil {
+		log.Errorf("DestroyModule RPC failed with err: %v\n", err)
+		return err
+	}
+
+	if resp.GetError() != nil && resp.GetError().Code != 0 {
+		log.Errorf("DestroyModule request '%+v' failed with err: %v\n", req, resp.GetError())
+		return status.Error(codes.Code(resp.GetError().Code), resp.GetError().Errmsg)
+	}
+
+	log.Tracef("Destroyed module %v", moduleName)
+
+	return nil
+}
+
+func (a *aether) sessionStats(pc *PfcpNodeCollector, ch chan<- prometheus.Metric) (err error) {
+	// TODO: implement
+	log.Traceln("sessionStats are not implemented in aether pipeline")
+	return
+}
+
+func (a *aether) setupBpfRules() (err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), Timeout)
+	defer cancel()
+
+	// Do not modify. Hard-coded gates from aether.bess pipeline.
+	const (
+		ueTrafficPassGate     = 0
+		signalTrafficPassGate = 8192 - 1 // MAX_GATE - 1
+	)
+
+	// Pass-through filter for GTPU UE traffic.
+	ueFilter := "ip and dst host " + a.ownIp.IP.String() + " and udp dst port 2152"
+	if err = a.addBpfRule(ctx, ueFilter, -ueTrafficPassGate, ueTrafficPassGate); err != nil {
```

Comment:
typo?


```suggestion
	if err = a.addBpfRule(ctx, ueFilter, ueTrafficPassGate, ueTrafficPassGate); err != nil {
```

Reply:
I feel there is an issue with this call to `addBpfRule`, `ueTrafficPassGate` is duplicated and `signalTrafficPassGate` is not used.

Reply:
I feel there is an issue with this call to `addBpfRule`, `ueTrafficPassGate` is duplicated and `signalTrafficPassGate` is not used.